src/headerssync.cpp

```diff
@@ -0,0 +1,318 @@
+#include <headerssync.h>
+#include <logging.h>
+#include <pow.h>
+#include <timedata.h>
+#include <util/check.h>
+
+//! Store a commitment to a header every HEADER_COMMITMENT_FREQUENCY blocks
+constexpr size_t HEADER_COMMITMENT_FREQUENCY = 571;
+//! Pull out headers for acceptance during redownload once our buffer exceeds this amount
+constexpr size_t REDOWNLOAD_HEADERS_THRESHOLD = 25*HEADER_COMMITMENT_FREQUENCY;
```

Comment:
@glozow `REDOWNLOAD_HEADERS_THRESHOLD` will (in optimal configurations) always be a multiple of `HEADER_COMMITMENT_FREQUENCY`. This could just have been written as just `= 14275;`, but that would make this multiple-of relation less clear.

If it's confusing, I would suggest instead doing:

```c++
constexpr size_t HEADER_COMMITMENT_FREQUENCY = 571;
constexpr size_t REDOWNLOAD_HEADERS_FACTOR = 25;
```

and then replace the current instances of `REDOWNLOAD_HEADERS_THRESHOLD` with `HEADER_COMMITMENT_FREQUENCY * REDOWNLOAD_HEADERS_FACTOR`.

Reply:
Thanks, done.